iosxr_test.py

```diff
@@ -0,0 +1,262 @@
+
+A tool to run some basic tests on a IOS XRv vagrant box, using pxssh module.
+
+This will verify both IOS XR Linux and IOS XR Console access.
+
+Is called from iosxr_iso2vbox post generation to verify sanity of
+the Virtualbox,
+
+Can also be run manually, like this:
+
+python iosxr_test.py iosxrv-fullk9-x64.box
+'''
+
+from __future__ import print_function
+import sys
+import pexpect
+from pexpect import pxssh
+import subprocess
+import argparse
+import os
+from iosxr_iso2vbox import run
+
+try:
+    raw_input
+except NameError:
+    raw_input = input
+
+# Some defaults
+terminal_type = 'ansi'
+linux_prompt = r"[#$]"
+xr_prompt = "[$#]"
+login_timeout = 10
+hostname = "localhost"
+username = "vagrant"
+password = "vagrant"
+linux_port = 2222
+xr_port = 2223
+
+
+def bringup_vagrant(input_box):
+    '''
+    Bring up a vagrant box.
+
+    Clean up ssh keys from the generation of the virtualbox.
+
+    Use pxssh to fascilitate the ssh process.
+    '''
+    # Clean up Vagrantfile
+    try:
+        os.remove('Vagrantfile')
+    except OSError:
+        pass
+
+    print('Cleaning up ssh keys')
+    run('ssh-keygen -R [localhost]:2222')
+    run('ssh-keygen -R [localhost]:2223')
+
+    print("Bringing up '%s'..." % input_box)
+
+    run('vagrant init XRv64')  # Single node for now, in future could bring up two nodes and do more testing
+    run('vagrant box add --name XRv64 %s --force' % input_box)
+    run('vagrant up')
+
+    ports = str(subprocess.check_output('vagrant port', shell=True))
+    print(ports)
+
+    try:
+        s = pexpect.pxssh.pxssh()
+        s.login(hostname, username, password, terminal_type, linux_prompt, login_timeout, linux_port)
+    except pxssh.ExceptionPxssh, e:
+        print("pxssh failed on login")
+        print(str(e))
+
+
+def test_linux():
+    '''
+    Verify logging into IOS XR Linux.
+    Verify user is 'vagrant'.
+    Verify can ping 'google.com'.
+    Verify resolv.conf is populated.
+    '''
+    print('Testing XR Linux...')
+
+    try:
+        s = pxssh.pxssh()
+        s.login(hostname, username, password, terminal_type, linux_prompt, login_timeout, linux_port)
+
+        print('Check user:')
+        s.sendline('whoami')
+        result = s.expect(['vagrant', pexpect.EOF, pexpect.TIMEOUT])
+        if result == 0:
+            print('==>Correct user found')
+        elif result == 1:
+            print('EOF')
+            return False
+        elif result == 2:
+            print('Timed out - wrong user name')
+            return False
+        s.prompt()
+
+        print('Check pinging the internet:')
+        s.sendline("ping -c 4 google.com | grep '64 bytes' | wc -l")
+        result = s.expect(['4', pexpect.EOF, pexpect.TIMEOUT])
+        if result == 0:
+            print('==>Successfully pinged')
+        elif result == 1:
+            print('EOF')
+            return False
+        elif result == 2:
+            print('Timed out - ping failed')
+            return False
```

Comment:
Ack.

Added the else too - although it's a one for one - so I don't think it can get hit.